src/main/java/core/basesyntax/service/impl/operation/PurchaseOperation.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl.operation;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.Fruit;
+
+public class PurchaseOperation implements OperationHandler {
+    private final FruitStorageDao fruitStorageDao;
+
+    public PurchaseOperation(FruitStorageDao fruitStorageDao) {
+        this.fruitStorageDao = fruitStorageDao;
+    }
+
+    @Override
+    public boolean doOperation(Fruit fruit, int quantity) {
```

Comment:
Handling Purchase operation requires checking the result balance before saving it in the Storage - it should be positive. Make sure that after subtracting the quantity, the balance remains non-negative, and throw a RuntimeException if the balance would become negative.